tests/sentry/deletions/test_cleanup.py

```diff
@@ -0,0 +1,99 @@
+from sentry.models.group import Group, GroupStatus
+from sentry.models.grouphash import GroupHash
+from sentry.runner.commands.cleanup import _cleanup
+from sentry.services.eventstore.models import Event
+from sentry.testutils.cases import TransactionTestCase
+from sentry.testutils.helpers.datetime import before_now
+
+RETENTION_DAYS = 2
+LESS_THAN_RETENTION_DAYS = before_now(days=RETENTION_DAYS - 1).isoformat()
+MORE_THAN_RETENTION_DAYS = before_now(days=RETENTION_DAYS + 1).isoformat()
+
+
+# TransactionTestCase is needed to ensure that the events are stored in the same transaction
+# as the cleanup process since it uses raw SQL queries and cannot see test transactions
+class TestGroupDeletion(TransactionTestCase):
+    """Test the group deletion functionality."""
+
+    def create_group(
+        self,
+        fingerprint: str,
+        timestamp: str,
+        status: GroupStatus = GroupStatus.UNRESOLVED,
+        project_id: int | None = None,
+    ) -> Event:
+        if project_id is None:
+            project_id = self.project.id
+        event = self.store_event(
+            data={"fingerprint": [fingerprint], "timestamp": timestamp}, project_id=project_id
+        )
+        assert event.group is not None
+
+        if status != GroupStatus.UNRESOLVED:
+            event.group.status = status
+            event.group.substatus = None
+            event.group.save()
+
+        assert event.group.status == status
+
+        return event
+
+    def test_only_deletes_old_groups(self) -> None:
+        """Test cleanup only deletes old groups."""
+        young_event = self.create_group("group1", LESS_THAN_RETENTION_DAYS)
+        group_id = young_event.group.id
+        self.create_group("group2", MORE_THAN_RETENTION_DAYS)
+        assert Group.objects.count() == 2
+        assert GroupHash.objects.count() == 2
+
+        _cleanup(
+            model=("Group",),
+            days=RETENTION_DAYS,
+            router="default",
+            silent=True,
+            disable_multiprocessing=True,
+        )
+
+        assert Group.objects.filter(id=group_id).exists()
+        assert Group.objects.count() == 1
+        assert GroupHash.objects.count() == 1
+
+    def test_delete_old_pending_groups(self) -> None:
+        """Test cleanup does not delete pending deletion groups."""
+        self.create_group("group1", MORE_THAN_RETENTION_DAYS, GroupStatus.PENDING_DELETION)
+
+        assert GroupHash.objects.count() == 1
+        assert Group.objects.count() == 1
+
+        _cleanup(
+            model=("Group",),
+            days=RETENTION_DAYS,
+            router="default",
+            silent=True,
+            disable_multiprocessing=True,
+        )
+
+        assert Group.objects.count() == 0
```

Comment:
I was expecting this test not to delete pending_deletion groups. I will investigate further.